projects/aws/bottlerocket-bootstrap/pkg/utils/userdata.go

```diff
@@ -24,28 +27,89 @@ type WriteFile struct {
 	Content     string
 }
 
+type UserDataSource struct {
+	Provider string
```

Comment:
Would we have other provider than secrets-manager?

Reply:
Currently secrets-manager is the only one we support. I leave it there because CAPI also support storing user data using SSM parameter store, (even through we are not using it).
I'm open to either keep/remove that.